crypto/relic_build.sh

```diff
@@ -36,13 +36,15 @@ CC_VAL=${CC_VAL:-"$(which cc)"}
 CC_VERSION_STR="$($CC_VAL --version)"
 
 # we use uname to record which arch we are running on
-ARCH=$(uname -m 2>/dev/null ||true)
+ARCH=$(uname -m 2>/dev/null || true)
 
-if [[ "$ARCH" =~ ^(arm64|armv7|armv7s)$ && "${CC_VERSION_STR[0]}" =~ (clang)  ]]; then
-    #  the "-march=native" option is not supported with clang on ARM
-    MARCH=""
+# Compile x86_64 as westmere by default.
+# See https://github.com/onflow/flow-go/pull/1993#issuecomment-1056048192
+if [[ "$ARCH" = "x86_64" ]]; then
+    MARCH="-march=westmere"
```

Comment:
Let's keep "native" with ARM-gcc as it is in the original script, because I didn't investigate the 'native" impact in that case yet.

```suggestion
if [[ "$ARCH" =~ "x86_64" ]]; then
# Compile as westmere arch to avoid cross-compilation issues on machines not supporting AVX extensions.
# Relic performance as used in flow crypto library is not impacted by whether it is compiled with "native" or "westmere", as proven by benchmark results.  
    MARCH="-march=westmere"
elif [[ "$ARCH" =~ ^(arm64|armv7|armv7s)$ && "${CC_VERSION_STR[0]}" =~ (clang)  ]]; then
    #  the "-march=native" option is not supported with clang on ARM
    MARCH=""
else
    MARCH="-march=native"
```

Reply:
I think the default is `native` for gcc under ARM, which is ok.

I will commit your suggestion either way.